docs/docs/loading-data-asynchronously.md

```diff
@@ -0,0 +1,155 @@
+---
+id: loading-data-asynchronously
+title: Loading Data Asynchronously
+permalink: docs/loading-data-asynchronously.html
+---
+
+React has no special capabilities for dealing with asynchronous network requests and a third party library or browser API is needed to perform them. If a component needs to have its UI respond to new data arriving, it has to call `setState` to rerender itself.
+
+## Initial Render
+
+Often, the data that a component needs is not available at initial render. We can load data asynchronously in the [`componentDidMount` lifecycle hook](/react/docs/react-component.html#componentdidmount).
+
+In the following example we use the [axios](https://github.com/mzabriskie/axios) to retrieve information about Facebook's Repos on GitHub and store them in the state. So first install it:
+
+```
+$ npm install axios --save
+```
+
+And you can use it in your project:
+
+```javascript{1,9-13}
+import axios from 'axios';
+
+class Repos extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {repos: []};
+  }
+
+  componentDidMount() {
+    axios.get('https://api.github.com/users/facebook/repos')
+      .then(response => this.setState({ repos: response.data }))
+      .catch(error => console.log(error));
+  }
+
+  render() {
+    return (
+      <div>
+        <h1>Repos by facebook</h1>
+        {this.state.repos.map(repo =>
+          <div key={repo.id}>{repo.name}</div>
+        )}
+      </div>
+    );
+  }
+}
+```
+
+The component will perform an initial render without any of the network data. When the axios promise resolves, it calls `setState` and the component is rerendered.
+
+[Try it on CodePen.](http://codepen.io/dashtinejad/pen/wgzEXJ?editors=0011)
+
+## Updates
+
+If the props change, we might need to fetch new data for the updated props. The [`componentDidUpdate` lifecycle hook](/react/docs/react-component.html#componentdidupdate) is a good place to achieve this, since we may not need to fetch new data if the props that we're interested in have not changed.
+
+Building on the previous example, we will pass the username as a prop instead and fetch new repos when it changes:
+
+```javascript{7-11,,13-15,17-21,35,38,44-47,53}
+class Repos extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {repos: []};
+  }
+
+  fetchRepos() {
+    axios.get(`https://api.github.com/users/${this.props.username}/repos`)
+      .then(response => this.setState({ repos: response.data }))
+      .catch(error => console.log(error));
+  }
+
+  componentDidMount() {
+    this.fetchRepos();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.username != prevProps.username) {
```

Comment:
Won't this lead to, if you first load repos for foo, and then for bar, on the immediate load of bar it will show the foo repos under a header that says "bar repo"?